components/raftstore/src/store/peer.rs

```diff
@@ -4796,6 +4799,13 @@ where
             let read_ts = decode_u64(&mut req.get_header().get_flag_data()).unwrap();
             let safe_ts = self.read_progress.safe_ts();
             if safe_ts < read_ts {
+                let now = Instant::now();
```

Comment:
It's neither a cheap call. Maybe you can check if safe_ts is way less than read_ts first.